java/arcs/android/devtools/DevWebServerImpl.kt

```diff
@@ -14,35 +14,63 @@ package arcs.android.devtools
 import arcs.core.util.TaggedLog
 import fi.iki.elonen.NanoHTTPD
 import fi.iki.elonen.NanoWSD
+import fi.iki.elonen.NanoWSD.WebSocketFrame.CloseCode
 import java.io.IOException
+import java.lang.Exception
 
 /**
  * An extension of [NanoWSD] for devtools to connect Arcs to a remote device for debugging.
  */
-class DevWebSocket : NanoWSD(12345) {
-    private var wsdSocket: WsdSocket? = null
+object DevWebServerImpl : DevWebServer, NanoWSD("localhost", 33317) {
+
+    private var wsdSockets = mutableListOf<WsdSocket>()
```

Comment:
Nit: should be a val.. also a Set might be a better fit here, given that you do `remove(socket)` later down.